crypto.go

```diff
@@ -26,13 +28,21 @@ type dhScheme interface {
 	ID() KEMID
 	GenerateKeyPair(rand io.Reader) (KEMPrivateKey, KEMPublicKey, error)
 	Marshal(pk KEMPublicKey) []byte
+	MarshalPrivate(sk KEMPrivateKey) []byte
 	Unmarshal(enc []byte) (KEMPublicKey, error)
+	UnmarshalPrivate(enc []byte) (KEMPrivateKey, error)
 	DH(priv KEMPrivateKey, pub KEMPublicKey) ([]byte, error)
 	PublicKeySize() int
+
+	setEphemeralKeyPair(sk KEMPrivateKey, pk KEMPublicKey)
+	ephemeralPublic() KEMPublicKey
+	ephemeralPrivate() KEMPrivateKey
```

Comment:
Are these needed?  Don't see any calls, and they seem duplicative of `getEphemeralKeyPair`

Reply:
Probably not!